internal/cri/server/container_stats_list.go

```diff
@@ -179,6 +179,17 @@ func (c *criService) toCRIContainerStats(css []containerStats) *runtime.ListCont
 }
 
 func (c *criService) getUsageNanoCores(containerID string, isSandbox bool, currentUsageCoreNanoSeconds uint64, currentTimestamp time.Time) (uint64, error) {
+	// First, try to get pre-calculated UsageNanoCores from the background stats collector.
+	// This ensures we have valid data even on the first query (as the collector runs
+	// continuously in the background, similar to cAdvisor's housekeeping).
+	if c.statsCollector != nil {
```

Comment:
should we check platform here given container stats collector is not implemented outside linux

Reply:
nod but with the original old model that we'll drop down to below that still runs for the other platforms .. (but not for sandbox as sandbox is no-op for other platforms in the older model...